app/grandchallenge/algorithms/models.py

```diff
@@ -427,17 +427,34 @@ def remove_user(self, user):
 
     def get_jobs_limit(self, user):
         """Get the maximum number of jobs a user can schedule now"""
-        if self.is_editor(user=user):
-            # Not limited by credits
-            return
+        if not self.active_image:
+            return None
+
+        editor_jobs = Job.objects.filter(
+            creator__in=self.editors_group.user_set.all(),
+            status=Job.SUCCESS,
+            algorithm_image__image_sha256=self.active_image.image_sha256,
+        ).values_list("pk", flat=True)
+        user_credit = Credit.objects.get(user=user)
+        credits_left = max(user_credit.credits, 0)
+        credits_per_job = max(self.credits_per_job, 1)
+
+        if (
+            self.is_editor(user=user)
+            and len(editor_jobs) < settings.ALGORITHMS_JOB_LIMIT_FOR_EDITORS
+        ):
+            return (
+                (credits_left // credits_per_job)
+                + settings.ALGORITHMS_JOB_LIMIT_FOR_EDITORS
+                - len(editor_jobs)
+            )
         else:
-            user_credit = Credit.objects.get(user=user)
-            jobs = Job.objects.spent_credits(user=user)
+            jobs = Job.objects.exclude(
+                pk__in=editor_jobs[: settings.ALGORITHMS_JOB_LIMIT_FOR_EDITORS]
```

Comment:
If we were to exclude all editor jobs here, all editors would always end up with the full amount of credits (i.e. nothing spent). And it doesn't matter which ones we exclude, right? All jobs cost equally much. 

But I think your suggestion to increase the editors user credits rather than decreasing the spent credits is easier to follow, so I will adopt that approach.